.github/workflows/build.yml

```diff
@@ -0,0 +1,89 @@
+name: 'confluent-kafka-dotnet build pipeline'
+
+env:
+  DOTNET_CLI_TELEMETRY_OPTOUT: 'true'
+
+on:
+  push:
+  pull_request:
+
+jobs:
+
+  build-test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13] # macos-13 for x86_x64 arch
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: | 
+              6.0.x
+              8.0.x
+      - name: Build and test
+        run: |
+          dotnet nuget add source  --username user --password ${{ github.token }} --store-password-in-clear-text --name github https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json
```

Comment:
could you explain why this is necessary? I had a brief look at v2.8.0-gr and we didn't seem to need this.

Reply:
In the `v2.8.0-gr`, we didn't restore the right version (because the GiHub package registry wasn't added as a package source). Thus, we've run tests against the official build instead:
```
warning NU1603: Confluent.Kafka depends on librdkafka.redist (>= 2.10.0-gr) but librdkafka.redist 2.10.0-gr was not found. librdkafka.redist 2.10.0-RC2 was resolved instead. [D:\a\confluent-kafka-dotnet\confluent-kafka-dotnet\Confluent.Kafka.sln]
```